content/en/platform/corda/5.0/developing-applications/api/ledger/utxo-ledger/vault-queries.md

```diff
@@ -0,0 +1,558 @@
+                        "->> 'testField' = :testField"
+            )
+            .filter(DummyCustomQueryFilter())
+            .map(DummyCustomQueryMapper())
+            .collect(DummyCustomQueryCollector())
+            .register()
+    }
+}
+```
+
+{{% /tab %}}
+{{% tab name="java" %}}
+
+```java
+public class JsonQueryFactory implements VaultNamedQueryFactory {
+    @Override
+    public void create(@NotNull VaultNamedQueryBuilderFactory vaultNamedQueryBuilderFactory) {
+        vaultNamedQueryBuilderFactory.create("DUMMY_CUSTOM_QUERY")
+                .whereJson(
+                        "WHERE visible_states.custom_representation -> 'com.r3.corda.demo.contract.TestState' " +
+                                "->> 'testField' = :testField"
+                )
+                .filter(new DummyCustomQueryFilter())
+                .map(new DummyCustomQueryMapper())
+                .collect(new DummyCustomQueryCollector())
+                .register();
+    }
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+{{< note >}}
+The collector always needs to be the last one in the chain as all previous filtering and mapping functionality will be applied before collecting.
+{{< /note >}}
+
+### How to Execute a Vault Named Query
+
+To execute a query use `UtxoLedgerService`. This can be injected to a flow via `@CordaInject`.
+To instantiate a query call the following:
+
+```kotlin
+utxoLedgerService.query("DUMMY_CUSTOM_QUERY", Int::class.java)
+```
+
+```java
+utxoLedgerService.query("DUMMY_CUSTOM_QUERY", Integer.class)
+```
+
+Provide the name of the query (in this case `DUMMY_CUSTOM_QUERY`) and the return type. Since the result set is collected into an integer in the complex query example use `Int` (or `Integer` in Java).
+
+Before executing define the following:
+
+* Which index the result set should start (`setOffset`), default 0.
+* How many results should the query return (`setLimit`), default Int.MAX (2,147,483,647).
+* Define named parameters that are in the query and the actual value for them. {{< note >}} All parameters must be defined otherwise the execution will fail. (`setParameter` or `setParameters`) {{</ note >}}
+* Each state in the database has a timestamp value for when it was inserted. Set an * upper limit to only return states that were inserted before a given time. (`setTimestampLimit`)
+
+    In this case it would look like this:
+
+    {{< tabs name="tabs-7" >}}
+    {{% tab name="kotlin" %}}
+
+    ```kotlin
+    val resultSet = utxoLedgerService.query("DUMMY_CUSTOM_QUERY", Int::class.java) // instantiate the query
+                .setOffset(0) // Start from the beginning
+                .setLimit(1000) // Only return 1000 records
+                .setParameter("testField", "dummy") // Set the parameter to a dummy value
+                .setCreatedTimestampLimit(Instant.now()) // Set the timestamp limit to the current time
+                .execute() // execute the query
+    ```
+
+    {{% /tab %}}
```

Comment:
Hard to tell in this format but the colour of the code from here on suggests something might be incorrect with your indentation or closing of code..